nikola/plugins/compile/markdown/__init__.py

```diff
@@ -43,40 +44,56 @@
 from nikola.utils import makedirs, req_missing, write_metadata
 
 
+class ThreadLocalMarkdown(threading.local):
+    """Convert to markdown using per-thread Markdown objects."""
+
+    def __init__(self, extensions):
+        self.markdown = Markdown(extensions=extensions, output_format="html5")
```

Comment:
With an integer it wouldn't be a problem anyway, because integers are immutable. If what I first thought would happen would really happen, you'd need a list or some other mutable object to test for it.

After reading the docstring for `_threading_local`, I'm happy with the code. I still think that `threading.local` viloates the principle of least astonishment.